src/iexamine.cpp

```diff
@@ -3431,6 +3432,144 @@ void iexamine::climb_down( player &p, const tripoint &examp )
+ * 
+ * TODO: Finish writting. Change to uimenu.
+ */
+void iexamine::base_control( player &p, const tripoint &examp) {
+    #define base (p->home) 
+    if (base == NULL){
+        add_msg("The GM prevents you from messing with other people's base settings.");
+        return;
+    }
+    
+    uimenu desktop;
+    desktop.text = _("Base Command Options:");
+    desktop.addentry( 1, true, MENU_AUTOASSIGN, _("Ration Supplies.") );
+    desktop.addentry( 2, true, MENU_AUTOASSIGN, _("Ammunition Type Prefrences.") );
+    desktop.addentry( 3, true, MENU_AUTOASSIGN, _("Crafting Priorities.") );
+    if (base.get_level() > 1){//add lv.2 options
+        desktop.addentry( 4, true, MENU_AUTOASSIGN, _("View Status.") );
+        desktop.addentry( 5, true, MENU_AUTOASSIGN, _("Base Planner.") )
+        if (base.get_level() > 2){//add lv.3 options
+            desktop.addentry( 6, true, MENU_AUTOASSIGN, _("Personel Managment.") );
+            if (base.get_level() > 3){//lv.4 allows basic access to aux system options
+            }
+        }
+    }
+    desktop.addentry( 0, true, MENU_AUTOASSIGN, _("Cancel") ); //cancel always 0 and at bottom
+    //desktop.addentry( , true, MENU_AUTOASSIGN,  );
+    
+    int topchoice=0;
+    desk_menu:
+    desktop.query();
+    deskchoice = desktop.ret;
+
+    switch (deskchoice) {
+        case 0: return;//cancel -> done
+        case 1: {//rationing rules
+            uimenu ratrule;
+
+            int ratwhat = menu(true, _("Food Rationing..."),
+                _("Ammunition Rationing..."),
+                _("Meds/First Aid Rationing..."),
+                _("Back"), NULL);
+            switch (ratwhatk){
+                case 1: {
+                    int lv = menu(true, _("Ration Level:"),
+                        _("None: Eat anything and everything whenever you want."),
+                        _("Light: Eat when hungry. Pay some attention to food use."),
+                        _("Medium: Eat when hungry. Always cook nutritious food."),
+                        _("Heavy: Eat when very hungry. Optimize food usage."), NULL);
+                    p.home.set_food_ration(lv-1);
+                }
+                case 2: {
+                    int lv = menu(true, _("Ration Level:"),
+                        _("None: Use anything and everything you want."),
+                        _("Light:"),
+                        _("Medium:"),
+                        _("Heavy:"), NULL);
+                    p.home.set_ammo_ration(lv-1);
+                }
+                case 3: {
+                    int lv = menu(true, _("Ration Level:"),
+                        _("None: Use anything and everything you want."),
+                        _("Light:"),
+                        _("Medium:"),
+                        _("Heavy:"), NULL);
+                    p.home.set_med_ration(lv-1);
+                }
+                case 4: {
+                    goto top_menu;
```

Comment:
Please don't use `goto`, ever. It considered to be one of the most awful features of the language. Seriously, google "c++ goto" and forget that it exists.